### PR TITLE
[FLINK-22492][test] Use order-agnostic check in KinesisTableApiITCase.

### PR DESCRIPTION
**Pull request by AHeise**
## What is the purpose of the change

KinesisTableApiITCase may fail as the query doesn't impose any ordering and a reorder may happen.

## Brief change log

- Use order-agnostic check in KinesisTableApiITCase.

## Verifying this change

Test fix.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-end-to-end-tests/flink-streaming-kinesis-test/src/test/java/org/apache/flink/streaming/kinesis/test/KinesisTableApiITCase.java [hunk 1]
```diff
@@ -108,7 +109,7 @@ public void testTableApiSourceAndSink() throws Exception {
         executeSqlStatements(readSqlFile("filter-large-orders.sql"));
 
         List<Order> result = readAllOrdersFromKinesis(kinesisClient);
-        assertEquals(expected, result);
+        assertThat(result, Matchers.containsInAnyOrder(expected.toArray(new Order[0])));
```
Comment (fapaul) [inline flink-end-to-end-tests/flink-streaming-kinesis-test/src/test/java/org/apache/flink/streaming/kinesis/test/KinesisTableApiITCase.java:112]: Maybe you can add a comment here that the order is not guaranteed.

File: flink-end-to-end-tests/flink-streaming-kinesis-test/src/test/java/org/apache/flink/streaming/kinesis/test/model/Order.java [hunk 2]
```diff
@@ -57,4 +57,9 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(code, quantity);
     }
+
+    @Override
+    public String toString() {
```
Comment (fapaul) [inline flink-end-to-end-tests/flink-streaming-kinesis-test/src/test/java/org/apache/flink/streaming/kinesis/test/model/Order.java:62]: I suppose this change was only needed for debugging purposes?

**Review by fapaul — APPROVED** [review]
Nice catch, thanks for fixing this.

File: flink-end-to-end-tests/flink-streaming-kinesis-test/src/test/java/org/apache/flink/streaming/kinesis/test/model/Order.java (hunk 2 above)
Comment (AHeise) [reply flink-end-to-end-tests/flink-streaming-kinesis-test/src/test/java/org/apache/flink/streaming/kinesis/test/model/Order.java:62]: `toString` is actually always needed for assertions ;)

**Comment by AHeise** [comment]
Failure unrelated (FLINK-22818) merging.

**Pull request merged**